modules/network/src/main/java/org/apache/ignite/internal/network/recovery/RecoveryClientHandshakeManager.java

```diff
@@ -136,14 +158,35 @@ public void onMessage(NetworkMessage message) {
             this.remoteLaunchId = msg.launchId();
             this.remoteConsistentId = msg.consistentId();
 
-            this.recoveryDescriptor = recoveryDescriptorProvider.getRecoveryDescriptor(
+            RecoveryDescriptor descriptor = recoveryDescriptorProvider.getRecoveryDescriptor(
                     remoteConsistentId,
                     remoteLaunchId,
-                    connectionId,
-                    false
+                    connectionId
             );
 
-            handshake(recoveryDescriptor);
+            while (!descriptor.acquire(ctx)) {
+                if (launchId.compareTo(remoteLaunchId) > 0) {
```

Comment:
Could you please add a comment about this comparison (or extract it to a method with a clarifying name)?

Reply:
Done